rotkehlchen/db/dbhandler.py

```diff
@@ -2233,6 +2233,66 @@ def get_netvalue_data(
                 data.append(total)
         return times_int, data
 
+    @staticmethod
+    def _infer_zero_timed_balances(
+            cursor: 'DBCursor',
+            balances: list[SingleDBAssetBalance],
+            from_ts: Optional[Timestamp] = None,
+            to_ts: Optional[Timestamp] = None,
+    ) -> list[SingleDBAssetBalance]:
+        """
+        Given a list of asset specific timed balances, infers the missing zero timed balances
+        for the asset.
+
+        Example
+        We have the following timed balances for ETH (value, time):
+        (1, 1), (1, 2), (2, 3), (5, 6), (5, 9)
+        The timestamps of all timed balances in the DB are:
+        (1, 2, 3, 4, 5, 6, 7, 8, 9)
+        So we need to infer the following zero timed balances:
+        (0, 4), (0, 7)
+        """
+        if len(balances) == 0:
+            return []
+
+        # else
+        cursor.execute(
+            'SELECT COUNT(*) FROM timed_balances WHERE timestamp BETWEEN ? AND ?',
+            (from_ts, to_ts),
+        )
+        num_all_balances = cursor.fetchone()[0]
+
+        if len(balances) == num_all_balances:
+            return []
+
+        # else
+        asset_timestamps = [b.time for b in balances]
+        entries = cursor.execute(
+            'SELECT timestamp, category FROM timed_balances WHERE timestamp BETWEEN ? AND ? '
+            'ORDER BY timestamp ASC',
+            (from_ts, to_ts),
+        ).fetchall()
+        all_timestamps = [e[0] for e in entries]
+        all_categories = [BalanceType.deserialize_from_db(e[1]) for e in entries]
+        # dicts maintain insertion order in python 3.7+
+        timestamps_dict = dict.fromkeys(all_timestamps, False)
+        asset_dict = dict.fromkeys(asset_timestamps, True)
+        timestamps_dict.update(asset_dict)
+        prev_from_asset = timestamps_dict[all_timestamps[0]]  # the value of the first timestamp
```

Comment:
I believe it was a bit harder to follow due to the names used for the variables? not sure if for you is the same. What do you think of

```suggestion
        timestamps_had_balance = dict.fromkeys(all_timestamps, False)
        timestamps_with_asset_balance = dict.fromkeys(asset_timestamps, True)
        timestamps_had_balance.update(asset_dict)
        prev_from_asset = timestamps_had_balance[all_timestamps[0]]  # the value of the first timestamp
```